substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/code/CodeInfoEncoder.java

```diff
@@ -393,6 +577,286 @@ public boolean verifyFrameInfo(CodeInfo info) {
+                        }
+                    } catch (IllegalAccessException | InvocationTargetException e) {
+                        return false;
+                    }
+                }
+            } else if (type.isArray()) {
+                boolean supported = true;
+                Class<?> componentType = type.getComponentType();
+                if (!componentType.isPrimitive()) {
+                    for (Object val : (Object[]) value) {
+                        supported &= supportedValue(componentType, val, stringValues);
+                    }
+                } else if (componentType == boolean.class) {
+                    for (boolean val : (boolean[]) value) {
+                        supported &= supportedValue(componentType, val, stringValues);
+                    }
+                } else if (componentType == byte.class) {
+                    for (byte val : (byte[]) value) {
+                        supported &= supportedValue(componentType, val, stringValues);
+                    }
+                } else if (componentType == short.class) {
+                    for (short val : (short[]) value) {
+                        supported &= supportedValue(componentType, val, stringValues);
+                    }
+                } else if (componentType == char.class) {
+                    for (char val : (char[]) value) {
+                        supported &= supportedValue(componentType, val, stringValues);
+                    }
+                } else if (componentType == int.class) {
+                    for (int val : (int[]) value) {
+                        supported &= supportedValue(componentType, val, stringValues);
+                    }
+                } else if (componentType == long.class) {
+                    for (long val : (long[]) value) {
+                        supported &= supportedValue(componentType, val, stringValues);
+                    }
+                } else if (componentType == float.class) {
+                    for (float val : (float[]) value) {
+                        supported &= supportedValue(componentType, val, stringValues);
+                    }
+                } else if (componentType == double.class) {
+                    for (double val : (double[]) value) {
+                        supported &= supportedValue(componentType, val, stringValues);
+                    }
+                }
+                return supported;
+            } else if (type == Class.class) {
+                return encoders.sourceClasses.contains((Class<?>) value);
+            } else if (type == String.class) {
+                if (stringValues != null) {
+                    stringValues.add((String) value);
+                }
+            } else if (type.isEnum()) {
+                if (stringValues != null) {
+                    stringValues.add(((Enum<?>) value).name());
+                }
+                return encoders.sourceClasses.contains(type);
```

Comment:
When this returns `false`, there is no point in adding to `stringValues` beforehand. But if I interpret the code correctly, then `stringValues` will be ignored anyway in that case?

Reply:
Yes, `stringValues` collects strings during the check and only registers them at the end if it succeeds.